From 4a699201b54b1ebf3ee60a9a38365309b055038b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sergi=20Blanch-Torn=C3=A9?= <sergi@calcurco.cat>
Date: Tue, 23 May 2017 11:35:47 +0200
Subject: [PATCH] make python3 compatible the CommandParser().

---
 hwinfo/util/__init__.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/hwinfo/util/__init__.py b/hwinfo/util/__init__.py
index 510b6bd..100a2bf 100644
--- a/hwinfo/util/__init__.py
+++ b/hwinfo/util/__init__.py
@@ -52,7 +52,7 @@ def parse_item(self, item):
             matches = [m.groupdict() for m in re.finditer(regex, item)]
             mdicts = combine_dicts(matches)
             if mdicts:
-                rec = dict(rec.items() + mdicts.items())
+                rec = dict(list(rec.items()) + list(mdicts.items()))
         return rec
 
     def parse_items(self):
@@ -60,7 +60,7 @@ def parse_items(self):
             return [self.parse_item(self.DATA)]
         else:
             recs = []
-            for data in self.DATA.split(self.ITEM_SEPERATOR):
+            for data in self.DATA.decode().split(self.ITEM_SEPERATOR):
                 rec = self.parse_item(data)
                 recs.append(rec)
             return recs
